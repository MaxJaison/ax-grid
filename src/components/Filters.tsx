import { Box, Button, Select, Text } from '@chakra-ui/react';
import { ENERGY_TYPES } from '../utils/constants.ts';
import { FiltersProps } from '../utils/types.ts';
import { ChangeEvent } from 'react';

export const Filters = ({ onOpen, socket }: FiltersProps) => {
    return (
        <Box display="flex" alignItems="center" bg="#DFEAFF" px="5" py="3">
            <Text fontSize="18" pr="5">
                Select which type of energy to display in the grid:{' '}
            </Text>
            <Select w="130px" variant="filled" borderColor="#3182ce" onChange={(event: ChangeEvent<HTMLSelectElement>) => socket.emit('new-energy', event.target.value)} defaultValue={ENERGY_TYPES[0]}>
                {ENERGY_TYPES.map((energy: string, index: number) => {
                    return (
                        <option key={index} value={energy}>
                            {energy}
                        </option>
                    );
                })}
            </Select>
            <Button colorScheme="blue" ml="auto" onClick={onOpen}>
                Add New Deal
            </Button>
        </Box>
    );
};
