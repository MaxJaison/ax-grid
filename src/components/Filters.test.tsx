import { describe, expect, it } from 'vitest';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/vitest';
import { Filters } from './Filters.tsx';
import { Socket as ClientSocket } from 'socket.io-client';

describe('Filters Component', () => {
    it('renders the Filters component', async () => {
        render(<Filters onOpen={() => {}} socket={ClientSocket} />);

        expect(screen.getByText(/Select which type of energy to display in the grid/i)).toBeInTheDocument();
    });

    it('create Filters snapshot', async () => {
        const { asFragment } = render(<Filters onOpen={() => {}} socket={ClientSocket} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
