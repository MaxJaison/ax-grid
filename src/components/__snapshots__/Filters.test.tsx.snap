// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Filters Component > create Filters snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1jr4x0c"
  >
    <p
      class="chakra-text css-bxx4ns"
    >
      Select which type of energy to display in the grid: 
    </p>
    <div
      class="chakra-select__wrapper css-18sl0nf"
    >
      <select
        class="chakra-select css-1j3r4nt"
      >
        <option
          selected=""
          value="Solar"
        >
          Solar
        </option>
        <option
          value="Gas"
        >
          Gas
        </option>
        <option
          value="Wind"
        >
          Wind
        </option>
        <option
          value="Hydro"
        >
          Hydro
        </option>
        <option
          value="Kinetic"
        >
          Kinetic
        </option>
        <option
          value="Thermal"
        >
          Thermal
        </option>
      </select>
      <div
        class="chakra-select__icon-wrapper css-162mkon"
      >
        <svg
          aria-hidden="true"
          class="chakra-select__icon"
          focusable="false"
          role="presentation"
          style="width: 1em; height: 1em; color: currentColor;"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <button
      class="chakra-button css-51wpq4"
      type="button"
    >
      Add New Deal
    </button>
  </div>
</DocumentFragment>
`;
